src/study/system-model/component.cpp

```diff
@@ -96,7 +96,7 @@ ComponentBuilder& ComponentBuilder::withModel(const Model* model)
  * \return Reference to the ComponentBuilder object.
  */
 ComponentBuilder& ComponentBuilder::withParameterValues(
-  std::map<std::string, double> parameter_values)
+  std::map<std::string, Expressions::Visitors::ContextParameter> parameter_values)
```

Comment:
what do you mean by the name "ContextParameter"? I find it confusing

Reply:
renamed ParameterTypeAndValue